index.js

```diff
@@ -10,8 +10,42 @@ const COLORS = 'https://nt-cdn.s3.amazonaws.com/colors.json';
  * @param compHex filter for complementary color hex code
  * @returns Promise
  */
-const fetchColors = ({ name, hex, compName, compHex }) => {
-  throw Error('Not implemented');
+const fetchColors = async ({ name, hex, compName, compHex }) => {
+  try {
+    const response = await fetch(COLORS);
+    if (!response.ok) {
+      throw new Error(`Response status: ${response.status}`);
+    }
+
+    const json = await response.json();
+    if (name) {
+      const lowerCaseName = name.toLowerCase();
+      const result = json.filter(
+        (color) => color.name.toLowerCase().includes(lowerCaseName)
+      );
+      return result;
+    } else if (hex) {
+      const result = json.filter((color) => color.hex === hex);
+      return result;
+    } else if (compName) {
+      const lowerCaseCompName = compName.toLowerCase();
+      const result = json.filter((color) =>
+        color.comp.some(
+          (compColor) => compColor.name.toLowerCase().includes(lowerCaseCompName)
+        )
+      );
+      return result;
+    } else if (compHex) {
+      const result = json.filter((color) =>
```

Comment:
no need to create a `result` variable if all it does is return. you can return directly.